### PR TITLE
Add logr v0.7.0

### PR DESCRIPTION
**Pull request by ngrodzitski**
Specify library name and version:  **logr/0.7.0**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ngrodzitski** [comment]
> Hello @ngrodzitski ! Thank you for updating this package. Is it not possible to re-use the previous recipe version in `all` folder with cmake commands?

I would prefer not to do it. As previous cmake contained `LOGR_CONAN_PACKAGING` that tuned the behavior of projects' cmake. And simply copying headers for header-only lib looks nicer to me in the way conanfile looks.

File: recipes/logr/v0.7/conanfile.py
Comment (RubenRBS) [inline recipes/logr/v0.7/conanfile.py:22]: Missing `package_type = "header-library"` attribute

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Thanks a lot for the PR and your patience, we appreciate it :)

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [inline recipes/logr/v0.7/conanfile.py:35]:
```suggestion
        self.requires("fmt/10.1.1")
```

Please, revert. Version ranges is allowed only to few packages: https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#version-ranges


About your conflict version error, we will need to open a second PR to update other dependencies.

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [inline recipes/logr/v0.7/conanfile.py:29]: @ngrodzitski Looking the project, it seems like we can have more than one backend at same time. Is it correct? https://github.com/ngrodzitski/logr/blob/main/CMakeLists.txt#L39

Is this more aligned with the project's design? 

```suggestion
    options = {
        "with_spdlog_backend": [True, False],
        "with_glog_backend": [True, False],
        "with_log4cplus_backend": [True, False],
        "with_boostlog_backend": [True, False],
    }
    default_options = {
        "with_spdlog_backend": True,
        "with_glog_backend": True,
        "with_log4cplus_backend": True,
        "with_boostlog_backend": True,
    }
```

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [reply recipes/logr/v0.7/conanfile.py:22]: Done.

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [inline recipes/logr/v0.7/conanfile.py:105]:
```suggestion
        self.cpp_info.libdirs = []

```
- frameworkdirs is empty by default
- cmake_file_name uses the package name by default
- Both cmake_find_package and cmake_find_package_multi are not needed to new recipes (there is no test_v1_package in this tree)

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [inline recipes/logr/v0.7/conanfile.py:96]:
```suggestion
        include_folder = os.path.join(self.package_folder, "include", "logr")
        if not self.options.with_spdlog_backend:
            rm(self, "spdlog_backend.hpp", include_folder)

        if not self.options.with_glog_backend:
            rm(self, "glog_backend.hpp", include_folder)

        if not self.options.with_log4cplus_backend:
            rm(self, "log4cplus_backend.hpp", include_folder)

        if not self.options.with_boostlog_backend:
            rm(self, "boostlog_backend.hpp", include_folder)
```

f-string is not required here.

**Review by uilianries — COMMENTED** [review]
Looking good, only small changes.

File: recipes/logr/v0.7/conanfile.py
Comment (uilianries) [inline recipes/logr/v0.7/conanfile.py:53]:
```suggestion
            self.requires("spdlog/1.12.0", transitive_headers=True, transitive_lib=True)

        if self.options.with_glog_backend:
            self.requires("glog/0.6.0", transitive_headers=True, transitive_lib=True)

        if self.options.with_log4cplus_backend:
            self.requires("log4cplus/2.1.0", transitive_headers=True, transitive_lib=True)

        if self.options.with_boostlog_backend:
            self.requires("boost/1.83.0", transitive_headers=True, transitive_lib=True)
```

As a header-only project, public headers will consume dependencies, which will not be visible by default. Conan 2.x restricts external headers in case not asking to be public.

File: recipes/logr/v0.7/conanfile.py
Comment (RubenRBS) [inline recipes/logr/v0.7/conanfile.py:53]:
```suggestion
            self.requires("spdlog/1.12.0", transitive_headers=True, transitive_libs=True)

        if self.options.with_glog_backend:
            self.requires("glog/0.6.0", transitive_headers=True, transitive_libs=True)

        if self.options.with_log4cplus_backend:
            self.requires("log4cplus/2.1.0", transitive_headers=True, transitive_libs=True)

        if self.options.with_boostlog_backend:
            self.requires("boost/1.83.0", transitive_headers=True, transitive_libs=True)
```

File: recipes/logr/v0.7/conanfile.py
Comment (RubenRBS) [inline recipes/logr/v0.7/conanfile.py:21]:
```suggestion
    topics = ("logger", "development", "util", "utils", "header-only")
```

File: recipes/logr/v0.7/conanfile.py
Comment (RubenRBS) [inline recipes/logr/v0.7/conanfile.py:53]:
```suggestion
        if self.options.with_spdlog_backend:
            self.requires("spdlog/1.12.0")

        if self.options.with_glog_backend:
            self.requires("glog/0.6.0")

        if self.options.with_log4cplus_backend:
            self.requires("log4cplus/2.1.0")

        if self.options.with_boostlog_backend:
            self.requires("boost/1.83.0")
```

As per https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring, a `package_type = "header-library"` recipe will have those traits inferred by default :)

**Comment by RubenRBS** [comment]
Hopefully fixed the last remaining issues, thanks a lot for your patience :)

**Review by valgur — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]
LGTM pinging @RubenRBS to have a final review.

**Review by valgur — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Review by RubenRBS — CHANGES_REQUESTED** [review]
Oops

**Review by RubenRBS — DISMISSED** [review]
Fixed, approving now

**Review by RubenRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**
